assert/object_match.ts

```diff
@@ -52,15 +52,23 @@ function isObject(val: unknown): boolean {
   return typeof val === "object" && val !== null;
 }
 
+function defineProperty(target: object, key: PropertyKey, value: unknown) {
+  return Object.defineProperty(target, key, {
+    value,
+    configurable: true,
+    enumerable: true,
+    writable: true,
+  });
+}
+
 function filter(a: loose, b: loose): loose {
   const seen = new WeakMap();
   return filterObject(a, b);
 
   function filterObject(a: loose, b: loose): loose {
     // Prevent infinite loop with circular references with same filter
-    if ((seen.has(a)) && (seen.get(a) === b)) {
-      return a;
-    }
+    const memo = seen.get(a);
+    if (memo && (memo === b)) return a;
```

Comment:
Is this change necessary? What happens if memo is the value `false`?

Reply:
Just performance thing. If you read the context of the changed lines, you'll realize that will never happen.

Reply:
See these types: [cfbff35](https://github.com/denoland/std/pull/6342/commits/cfbff35b1f911ed32d1720a754f81d57f4ff1ebc)

If you say this change should be done in another PR, I understand it and revert this part, but I'd say such a trivial and small change can be okay to come along with...